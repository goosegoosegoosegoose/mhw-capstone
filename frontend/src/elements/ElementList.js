import React, { useState, useEffect } from "react";
import { Card, Button } from "react-bootstrap";
import { useNavigate, useParams } from "react-router-dom";
import MhwApi from "../api";
import ElementCard from "./ElementCard";

const ElementList = () => {
  const [elements, setElements] = useState([]);

  useEffect(function fetchElementsWhenMounted() {
    async function fetchElements() {
      const res = await MhwApi.request("elements");
      setElements(res)
    }
    fetchElements()
  }, []);

  if (!elements) {
    return (
      <div className="container text-center">
        <p>Loading</p>
      </div>
    )
  }

  return (
    <div className="container text-center">
      <div className="row justify-content-center">
        {elements.map(e => 
          <ElementCard
            key={e.element}
            ele={e.element}
          />
        )}
      </div>
  </div>
  )
}

export default ElementList;