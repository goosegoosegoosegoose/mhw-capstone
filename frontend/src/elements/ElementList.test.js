import { render } from '@testing-library/react';
import ElementList from "./ElementList";

test('smoke test', () => {
  render(      
    <ElementList />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<ElementList />);
  expect(asFragment()).toMatchSnapshot();
});