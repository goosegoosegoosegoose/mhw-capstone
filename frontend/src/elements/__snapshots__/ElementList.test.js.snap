// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="container text-center"
  >
    <h1>
      Elements
    </h1>
    <div
      class="row justify-content-center"
    />
  </div>
</DocumentFragment>
`;
